desc/vmec_utils.py

```diff
@@ -469,3 +487,509 @@ def vmec_boundary_subspace(eq, RBC=None, ZBS=None, RBS=None, ZBC=None):  # noqa:
+    lasym[:] = not eq.sym
+
+    # FIXME: ns?
+
+    ns = file.createVariable("ns_b", np.int32)
+    ns.long_name = "Number of radial surfaces at which data is outputted minus 1"
+    ns[:] = surfs
+
+    aspect = file.createVariable("aspect_b", np.float64)
+    aspect.long_name = "Aspect Ratio"
+    aspect[:] = eq.compute("R0/a")["R0/a"]
+
+    grid_lcfs = LinearGrid(M=eq.M_grid, N=eq.N_grid, rho=1, NFP=NFP)
+    Rs = eq.compute("R", grid=grid_lcfs)["R"]
+    Rmax = file.createVariable("rmax_b", np.int32)
+    Rmax.long_name = "Maximum Radius"
+    Rmax[:] = np.max(Rs)
+
+    Rmin = file.createVariable("rmin_b", np.int32)
+    Rmin.long_name = "Minimum Radius"
+    Rmin[:] = np.min(Rs)
+
+    # betaxis = beta_vol at the axis?
+    betaxis = file.createVariable("betaxis_b", np.float64)
+    betaxis[:] = 0
+    betaxis.long_name = "Not Implemented: This output is hard-coded to 0!"
+
+    mboz = file.createVariable("mboz_b", np.int32)
+    mboz.long_name = (
+        "Maximum poloidal mode number m for which the Fourier"
+        "amplitudes rmnc, bmnc etc are stored"
+    )
+    mboz[:] = M_booz
+
+    nboz = file.createVariable("nboz_b", np.int32)
+    nboz.long_name = (
+        "Maximum toloidal mode number m for which"
+        "the Fourier amplitudes rmnc, bmnc etc are stored"
+    )
+    nboz[:] = N_booz
+
+    mnboz = file.createVariable("mnboz_b", np.int32)
+    mnboz.long_name = (
+        "The total number of (m,n) pairs for which Fourier amplitudes"
+        "rmnc, bmnc etc are stored."
+    )
+    mnboz[:] = basis.num_modes
+
+    ## 1D Arrays
+    keys_1d = [
+        "I",
+        "G",
+    ]
+    # this should be compressed then to just the radial profiles
+    grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, rho=r_half, NFP=NFP)
+    data_1d = eq.compute(keys_1d, grid=grid)
+    jlist = file.createVariable("jlist", np.int32, ("comput_surfs",))
+    jlist.long_name = (
+        "1-based radial indices of the surfaces for which the "
+        "transformation to Boozer coordinates was computed. 2 corresponds to the "
+        "first half-grid point."
+    )
+    jlist.units = "None"
+    jlist[:] = np.arange(1, s_half.size + 1) + 1
+
+    ixm_b = file.createVariable("ixm_b", np.int32, ("mn_modes",))
+    ixm_b.long_name = (
+        "Poloidal mode numbers m for which the Fourier amplitudes rmnc,"
+        "bmnc etc are stored"
+    )
+    ixm_b.units = "None"
+    ixm_b[:] = modes[:, 1]
+
+    ixn_b = file.createVariable("ixn_b", np.int32, ("mn_modes",))
+    ixn_b.long_name = (
+        "Toroidal mode numbers n for which the Fourier amplitudes"
+        "rmnc, bmnc etc are stored"
+    )
+    ixn_b.units = "None"
+    ixn_b[:] = modes[:, 2] * eq.NFP
+
+    iotas = file.createVariable("iota_b", np.float64, ("radius",))
+    iotas.long_name = (
+        "Rotational transform. The radial grid corresponds to the"
+        "requested surfaces, and a 0 is prepended"
+    )
+    iotas.units = "None"
+    if eq.iota is not None:
```

Comment:
couldn't we just always use eq.compute to get iota?

Reply:
I copied these from the save vmec method, which at the time at least did not do this. I will update